src/datasets/iterable_dataset.py

```diff
@@ -719,6 +726,15 @@ class ShufflingConfig:
     generator: np.random.Generator
 
 
+def _maybe_add_torch_iterable_dataset_parent_class(cls):
+    """Add torch.utils.data.IterableDataset as a parent class if 'torch' is imported"""
```

Comment:
I think it happens even if `torch` is not imported but is installed in the env. Is it an intended behavior by the way?